src/net_processing.cpp

```diff
@@ -892,7 +893,13 @@ class PeerManagerImpl final : public PeerManager
      */
     bool BlockRequested(NodeId nodeid, const CBlockIndex& block, std::list<QueuedBlock>::iterator** pit = nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
-    bool TipMayBeStale() EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+    enum class Staleness {
```

Comment:
ef4a1a44dca1af20adcc8fb0c14e8a31683a9d99

nit: this enum could be better... First, `StalePast` also qualifies as `Stale` (just english-wise). Second, it's a negation (non-synced).

Reply:
Hmm, naming is hard.
As the function is called `TipMayBeStale()`, maybe calling it `SyncState` could also work. Thoughts? Or do you have anything else in mind?

Reply:
`SyncState` yeah that would work if you come up with a good enum values.

Reply:
> `SyncState` yeah that would work if you come up with a good enum values.

Maybe `Synced`, `RecentStale`, `Stale`?
Named the second and third "stale" because these states have no in-flight block requests.

Still, this could also be easily changed in the future if we come up with better names.